src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/HttpHandler.cs

```diff
@@ -0,0 +1,52 @@
+// <copyright file="HttpHandler.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Net.Http;
+using System.Threading;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation
+{
+    /// <summary>
+    /// Class decorating <see cref="System.Net.Http.HttpClient"/>.
+    /// </summary>
+    internal class HttpHandler : IHttpHandler
```

Comment:
Rather than creating a wrapper for `HttpClient` another approach is to use `HttpClient` directly and use a mock `HttpRequestHandler` from unit tests.

You can [construct](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.-ctor?view=net-5.0#System_Net_Http_HttpClient__ctor_System_Net_Http_HttpMessageHandler_) an `HttpClient` instance with an `HttpMessageHandler`.

Reply:
Got rid of this wrapper. Unit tests adapted respectively.